site/src/app/v2/test-eligibilite/components/radioButtonsGroup/RadioButtonsGroup.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import styles from './styles.module.scss';
+
+type Options = {
+  label: string;
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}[];
+
+interface Props {
+  fieldsetId: string;
+  options: Options;
+}
+
+const RadioButtonsGroup: React.FC<Props> = ({ options, fieldsetId }) => {
+  const [buttonClickedIndex, setButtonClickedIndex] = useState<number | null>(null);
+
+  const buttonHandler = (index: number) => {
+    setButtonClickedIndex(index);
+  };
+
+  const rewireOnClick = () => {
+    return options.map((option, index) => ({
+      ...option,
+
+      onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
+        buttonHandler(index);
+        option.onChange(e);
+      },
+    })) as Options;
+  };
```

Comment:
Je ne vois pas l'intérêt de cette function, autant mettre la propriété onChange directement en dessous lors de l'itération sur les options?

Reply:
oui, j'ai repris le mecanisme precedent. J'ai simplifié comme proposé